lib/rules/newline-after-var.js

```diff
@@ -114,6 +114,12 @@ module.exports = function(context) {
             return;
         }
 
+        // Ignore if it is last statement in a function
+        if (node.parent.parent && node.parent.parent.type === "FunctionDeclaration" &&
```

Comment:
This check isnt quite right. First, we need to check for all function forms (FunctionDeclaration, FunctionExpression, ArrowFunctionExpression).

Next, you can check that the last element in the body is equal to the VariableDeclaration.